Tuprules.tup

```diff
@@ -12,38 +12,11 @@ else
 CC = gcc
 endif
 
-ifeq (@(TUP_DEBUG),y)
-CFLAGS += -g
-else
-CFLAGS += -Os
-endif
+CFLAGS = @(CFLAGS)
 
-CFLAGS += -W
-CFLAGS += -Wall
-ifeq (@(TUP_WERROR),y)
-CFLAGS += -Werror
-endif
-CFLAGS += -Wbad-function-cast
```

Comment:
You are right. On the one hand, moving everything toolchain-specific to separate configs would allow to remove some ifdefs from Tupfiles. The sanctioned default could still be in tup.config so that this restructuration was transparent to users. But, on the other hand, having the default in tup.config would mean having it hard-coded, meaning that those cloning and building on Windows would get wrong flags by default. By having the default in Tupfiles, we are able to use conditional constructs to set sensible defaults depending on the target platform.

That said, @-variables like `CONFIG_TUP_WERROR` aren't needed. We don't really want to have a config variable for each compiler feature; it's just easier to let users specify those flags in their tup.config. `-Werror` and such should be moved to a template tup.config for developers (say, `dev.config`), while similar templates could be provided for release and LTO builds and whatnot.